atd-vze/src/views/Users/Users.js

```diff
@@ -11,15 +11,62 @@ import {
   Pagination,
   PaginationItem,
   PaginationLink,
+  Popover,
   Row,
   Table,
   Button,
   Spinner,
+  PopoverBody,
 } from "reactstrap";
 import Can from "../../auth/Can";
 import { useAuth0 } from "../../auth/authContext";
 import { rules } from "../../auth/rbac-rules";
 
+// Call the list_users API and return the users for the current page
+async function getCurrentPageUsers(page, perPage, token) {
+  const endpoint = `${process.env.REACT_APP_CR3_API_DOMAIN}/user/list_users?page=${page}&per_page=${perPage}`;
+  const res = await axios.get(endpoint, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+  return res;
+}
+
+// Fetch an array of all the users
+async function getAllUsers(pageCount, perPage, totalUsers, setAllUsers, token) {
+  let page = 0;
+  let users = [];
+  // query every page of users until we have fetched them all
+  while (page <= pageCount) {
+    const res = await getCurrentPageUsers(page, perPage, token);
+    res.data.users.forEach(user => {
+      // make sure the user is not blocked/inactive, then push to an array
+      if (user.status === false || user.status === undefined) {
+        users.push(user);
+      }
+    });
+    page++;
+  }
+  // once we have fetched all users, update the users array in state
+  if (users.length === totalUsers) {
+    setAllUsers(users);
+    return users;
+  }
```

Comment:
and if `users.length !== totalUsers` that would mean something weird happened with the Auth0 data fetching? from a UI perspective, would that just cause the loading spinner keep spinning forever? if this is an extreme edge case i could live with that, but at least log a console error?

Reply:
I was concerned about this, because if inactive users are in the list that is returned (and counted as part of total users) but then not added to the email list, then this wont work out. 

When I was testing I checked to see if any inactive users were returned in the users request, and they are not. every user is returned with an undefined status.

Reply:
Now that I'm thinking more about it though, I can see how even though this works now, all it would take would be someone to be in the list of users that is inactive and this feature breaks.

Reply:
good catch y'all. there are multiple places in the code that check if allUsers = totalUsers, which like chia said is gonna cause issues if there is ever an inactive user in totalUsers. thinking thru a solution now

Reply:
Okay im gonna just filter out inactive users in the `getUserEmails` function instead of filtering them out of `allUsers` state in `getAllUsers`